### ISSUE TITLE
MWAA 2.2.2 branch packaging apache airflow 2.2.3 whl file will auto upgrade MWAA to 2.2.3 in AWS env

### ISSUE THREAD

**Issue by shmcsensei** [aws/aws-mwaa-local-runner#281]
MWAA 2.2.2 branch packaging apache airflow 2.2.3 whl file will auto upgrade MWAA to 2.2.3 in AWS env.

When running `mwaa-local-env package-requirements` in the 2.2.2 branch we get the following
<img width="708" alt="Screenshot 2023-07-04 at 10 01 05 AM" src="https://github.com/aws/aws-mwaa-local-runner/assets/34966765/d7300c28-ec40-4778-8aae-fa0cacbc73c9">

as per the screen shot `apache_airflow-2.2.3-py3-none-any.whl` is packaged into the plugins.zip file which will automatically upgrade the AWS MWAA environment when uploaded.

This happened when checking out the 2.2.2 branch fresh and running `mwaa-local-env package-requirements` with no modifications.

**Comment by julio-tl** [comment]
@shmcsensei I encountered a similar issue this week, but this time around it brought in the wheel `apache_airflow-2.6.3-py3-none-any.whl`. This caused regression in our MWAA environment (Airflow 2.2.2), with pip installation failures and DAG import errors. I managed to resolve it with the following approach:

- Always include a `--constraints` reference in the requirements.txt file, regardless of the Airflow version.
- Point to [constraints-2.2.2/constraints-3.7.txt](https://raw.githubusercontent.com/apache/airflow/constraints-2.2.2/constraints-3.7.txt) or equivalent, which matches Airflow 2.2.2 and python 3.7
- Downgrade individual dependencies to resolve conflicts if any exist. I think that file already changed since yesterday, so in case it does again, here were the major differences I had to capture:
```
apache-airflow==2.2.2
apache-airflow-providers-amazon==2.4.0
PyYAML==5.3.1
watchtower==1.0.6
```
- The file in [aws-mwaa-local-runner](https://github.com/aws/aws-mwaa-local-runner/blob/v2.2.2/dags/constraints-3.7-amazon-6.0.txt) for local dependency validation is not guaranteed to be at parity with MWAA because it does not clamp down the Airflow version, we have to do it ourselves by adding `apache-airflow` to the list of requested dependencies. Perhaps there is a more elegant way to do this, but otherwise, there is no way for the tool to know which airflow version we are targetting.

I was very puzzled as to why this started happening. It turns out its been happening for a while, but it never caused issues until now. I checked the oldest `plugins.zip` file we have in S3 from several weeks ago and it already had an airflow 2.2.3 wheel in it... This must be caused by the depedency version requested by the amazon providers package. So if we request `apache-airflow-providers-amazon==2.4.0` the providers package will ask for `apache-airflow>=2.2.0` which can clearly lead to issues.
